src/ripple/nodestore/impl/DatabaseShardImp.cpp

```diff
@@ -144,40 +150,46 @@ DatabaseShardImp::init()
                     continue;
                 }
 
-                if (shard->isFinal())
-                {
-                    shards_.emplace(
-                        shardIndex,
-                        ShardInfo(std::move(shard), ShardInfo::State::final));
-                }
-                else if (shard->isBackendComplete())
+                switch (shard->getState())
                 {
-                    auto const result{shards_.emplace(
-                        shardIndex,
-                        ShardInfo(std::move(shard), ShardInfo::State::none))};
-                    finalizeShard(
-                        result.first->second, true, lock, boost::none);
-                }
-                else
-                {
-                    if (acquireIndex_ != 0)
-                    {
+                    case Shard::State::final:
+                        if (++openFinals > openFinalLimit_)
+                            shard->tryClose();
+                        shards_.emplace(shardIndex, std::move(shard));
+                        break;
+
+                    case Shard::State::complete:
+                        finalizeShard(
+                            shards_.emplace(shardIndex, std::move(shard))
```

Comment:
Are the move semantics necessary now that the shard is a `shared_ptr`? It doesn't hurt but it could be misleading to someone else reading the code.